File: app/src/main/java/com/mdaftabalam/imageslider/MainActivity.java
package com.mdaftabalam.imageslider;

import android.os.Handler;
import android.os.Bundle;
import android.view.MotionEvent;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import java.util.ArrayList;
import java.util.Timer;
import java.util.TimerTask;

public class MainActivity extends AppCompatActivity {

    private static ViewPager mPager;
    private static int currentPage = 0;
    private static int NUM_PAGES = 0;
    private ArrayList<SlidingModel> imageModelArrayList;

    private int[] myImageList = new int[]{R.drawable.banner1, R.drawable.banner2, R.drawable.banner3};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        imageModelArrayList = new ArrayList<>();
        imageModelArrayList = populateList();
        init();


    }

    private ArrayList<SlidingModel> populateList() {
        ArrayList<SlidingModel> list = new ArrayList<>();
        for (int i = 0; i < 3; i++) {
            SlidingModel imageModel = new SlidingModel();
            imageModel.setImage_drawable(myImageList[i]);
            list.add(imageModel);
        }
        return list;
    }

    private void init() {
        mPager = findViewById(R.id.pager);
        final View touchView = findViewById(R.id.pager);
        touchView.setOnTouchListener(new View.OnTouchListener()
        {
            @Override
            public boolean onTouch(View v, MotionEvent event)
            {
                return true;
            }
        });

        mPager.setAdapter(new SlidingAdapter(MainActivity.this, imageModelArrayList));
        NUM_PAGES = imageModelArrayList.size();



        // Auto start of viewpager
        final Handler handler = new Handler();
        final Runnable Update = new Runnable() {
            public void run() {
                if (currentPage == NUM_PAGES) {
                    currentPage = 0;
                }
                mPager.setCurrentItem(currentPage++, false);
            }
        };
        Timer swipeTimer = new Timer();
        swipeTimer.schedule(new TimerTask() {
            @Override
            public void run() {
                handler.post(Update);
            }
        }, 4000, 4000);
    }
}

File: app/src/main/java/com/mdaftabalam/imageslider/MainActivity.kt
package com.mdaftabalam.imageslider

import android.os.Bundle
import android.os.Handler
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager
import com.mdaftabalam.imageslider.MainActivity
import java.util.*

class MainActivity : AppCompatActivity() {
    private var imageModelArrayList: ArrayList<SlidingModel>? = null
    private val myImageList = intArrayOf(R.drawable.banner1, R.drawable.banner2, R.drawable.banner3)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        imageModelArrayList = ArrayList()
        imageModelArrayList = populateList()
        init()
    }

    private fun populateList(): ArrayList<SlidingModel> {
        val list = ArrayList<SlidingModel>()
        for (i in 0..2) {
            val imageModel = SlidingModel()
            imageModel.image_drawable = myImageList[i]
            list.add(imageModel)
        }
        return list
    }

    private fun init() {
        mPager = findViewById(R.id.pager)
        //disalble auto slide
//        val touchView = findViewById<View>(R.id.pager)
//        touchView.setOnTouchListener { v, event -> true }
        mPager!!.setAdapter(SlidingAdapter(this@MainActivity, imageModelArrayList!!))
        NUM_PAGES = imageModelArrayList!!.size


        // Auto start of viewpager
        val handler = Handler()
        val Update = Runnable {
            if (currentPage == NUM_PAGES) {
                currentPage = 0
            }
            mPager!!.setCurrentItem(currentPage++, false)
        }
        val swipeTimer = Timer()
        swipeTimer.schedule(object : TimerTask() {
            override fun run() {
                handler.post(Update)
            }
        }, 4000, 4000)
    }

    companion object {
        private var mPager: ViewPager? = null
        private var currentPage = 0
        private var NUM_PAGES = 0
    }
}